### ISSUE TITLE
Please catch vega exceptions

### ISSUE THREAD

**Issue by bashimao** [anibali/showoff#4]
Can we make so that vega does not kill the server if it is unhappy with the supplied data?

```
return dl.array(ref.field).map(function(f) {
web_1 |                      ^
web_1 | 
web_1 | TypeError: Cannot read property 'field' of undefined
web_1 |     at getFields (/app/node_modules/vega/src/scene/Scale.js:270:22)
web_1 |     at Node.getCache (/app/node_modules/vega/src/scene/Scale.js:306:16)
web_1 |     at Node.dataRef (/app/node_modules/vega/src/scene/Scale.js:360:24)
web_1 |     at Node.domainMinMax (/app/node_modules/vega/src/scene/Scale.js:429:15)
web_1 |     at Node.quantitative (/app/node_modules/vega/src/scene/Scale.js:223:18)
web_1 |     at Node.scale (/app/node_modules/vega/src/scene/Scale.js:89:5)
web_1 |     at fn (/app/node_modules/vega/src/scene/Scale.js:52:36)
web_1 |     at Array.forEach (native)
web_1 |     at Node.proto.evaluate (/app/node_modules/vega/src/scene/Scale.js:55:13)
web_1 |     at Model.prototype.evaluate (/app/node_modules/vega-dataflow/src/Graph.js:329:16)
web_1 |     at Model.prototype.propagate (/app/node_modules/vega-dataflow/src/Graph.js:161:22)
web_1 |     at Model.prototype.fire (/app/node_modules/vega/src/core/Model.js:159:8)
web_1 |     at View.prototype.update (/app/node_modules/vega/src/core/View.js:369:11)
web_1 |     at vega.js:11:47
web_1 |     at [object Object].done [as _onTimeout] (/app/node_modules/vega/src/parse/spec.js:97:26)
web_1 |     at Timer.listOnTimeout (timers.js:92:15)
```

**Comment by anibali** [comment]
Seems reasonable, I'll see what I can do

**Comment by anibali** [comment]
Fixed in 18994da2e9f58aa8ea2e9dd36007f523b687c4b7.
